Assets/Scripts/GameScene/PlayerController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerController : MonoBehaviour, IPlayerController
+{
+    [SerializeField] private Board board;
+    private PlayerInput _playerInput;
+    private CharacterController _characterController;
+    
+    
+    private Vector2 _movementInput;
+    private Vector3 _playerMovement;
+    private float _playerSpeed = 2.0f;
+
+    
+    private bool _isMovementPressed;
+    [SerializeField] private float rotationFactorPerFrame = 1.0f;
+    
+    
+    private void Awake()
+    {
+        InitPlayerInput();
+        _characterController = GetComponent<CharacterController>();
+    }
+    
+
+ 
+
+    private void Update()
+    {
+        HandleMovementWithPlayerInput();
+        // HandleMovement();
+        // _characterController.Move(_playerMovement * Time.deltaTime * _playerSpeed);
+    }
+
+    private void OnEnable()
+    {
+        _playerInput.PlayerContorls.Enable();
+    }
+
+    private void OnDisable()
+    {
+        _playerInput.PlayerContorls.Disable();
+    }
+
+    public Board GetGameBoard()
+    {
+        return AssetDatabase.LoadAssetAtPath<Board>("Assets/Data/Board.asset");
+    }
```

Comment:
Let's try to refrain from doing this since it creates a dependency between the script and specific file in the project, remove this function and get the board's scriptable object as a `[SerializeField] private...`